### PR TITLE
[spanify] Avoid `base::make_span`

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
This PR removes various uses of `base::make_span` across the codebase, in favour of the existing constructors that can be deduced with the use of CTAD.

Resolves https://github.com/brave/brave-browser/issues/42624

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by zenparsing — APPROVED** [review]
Rewards 👍

**Review by tmancey — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Review by fmarier — APPROVED** [review]
`components/brave_rewards/core/common/request_signer.cc` :+1:

**Review by goodov — APPROVED** [review]

File: components/brave_wallet/common/eth_abi_utils_unittest.cc [hunk 1]
```diff
@@ -175,7 +175,7 @@ TEST(EthAbiUtilsTest, ExtractAddress) {
   {
     auto bytes = ToBytes(
         "000000000000000000000000c1735677a60884abbcf72295e88d47764beda282");
-    EXPECT_EQ(ExtractAddress(base::make_span(bytes)).ToHex(),
+    EXPECT_EQ(ExtractAddress(base::span(bytes)).ToHex(),
```
Comment (supermassive) [inline components/brave_wallet/common/eth_abi_utils_unittest.cc:178]: nit: It seems `ExtractAddress(bytes)` would be enough

File: components/brave_rewards/core/common/request_signer_unittest.cc [hunk 2]
```diff
@@ -28,8 +28,7 @@ class RewardsRequestSignerTest : public testing::Test {
 
 TEST_F(RewardsRequestSignerTest, GetDigest) {
   auto seed = Signer::GenerateRecoverySeed();
-  auto digest =
-      RequestSigner::GetDigest(base::as_bytes(base::make_span("hello world")));
+  auto digest = RequestSigner::GetDigest(base::as_byte_span("hello world"));
   EXPECT_EQ(digest, "SHA-256=QwZGhH5wNEwJ9Yc56Z1byW6sjV/nKVzxlrmGJ5h2v5s=");
 }
```
Comment (supermassive) [inline components/brave_rewards/core/common/request_signer_unittest.cc:33]: It looks it like it should be `byte_span_from_cstring` which would change what this test tests but would be more correct.

File: components/brave_rewards/core/common/signer_unittest.cc [hunk 3]
```diff
@@ -34,7 +34,7 @@ TEST_F(RewardsSignerTest, SignMessage) {
             "ZsQSLeH43VfOCuzGBuFunm1cTEMafWN4zTY++3Oz6A==");
 
   auto signed_message = base::Base64Encode(
-      signer->SignMessage(base::as_bytes(base::make_span("hello world"))));
+      signer->SignMessage(base::as_byte_span("hello world")));
```
Comment (supermassive) [inline components/brave_rewards/core/common/signer_unittest.cc:37]: nit: Also worth changing to `byte_span_from_cstring` and EXPECTED string

**Review by supermassive — APPROVED** [review]
wallet core lgtm

File: components/brave_rewards/core/common/request_signer_unittest.cc (hunk 2 above)
Comment (cdesouza-chromium) [reply components/brave_rewards/core/common/request_signer_unittest.cc]: Fixed. Thanks. Using `byte_span_with_nul_from_cstring` to preserve the `nul` as in the example above.

File: components/brave_wallet/common/eth_abi_utils_unittest.cc (hunk 1 above)
Comment (cdesouza-chromium) [reply components/brave_wallet/common/eth_abi_utils_unittest.cc]: Fixed.

**Pull request merged**